seastar/src/config_and_start_seastar.rs

```diff
@@ -0,0 +1,339 @@
+        get_name(&self.opts)
+    }
+
+    /// Gets the `Options`' description.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let opts = Options::new();
+    ///
+    /// assert_eq!(opts.get_description(), "");
+    /// ```
+    pub fn get_description(&self) -> &str {
+        get_description(&self.opts)
+    }
+
+    /// Gets the `Options`' number of threads.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let opts = Options::new();
+    ///
+    /// assert_eq!(opts.get_smp(), num_cpus::get() as u32);
+    /// ```
+    pub fn get_smp(&self) -> u32 {
+        get_smp(&self.opts)
+    }
+
+    /// Sets the `Options`' name.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let mut opts = Options::new();
+    /// let name = "My Awesome Seastar App";
+    /// opts.set_name(name);
+    ///
+    /// assert_eq!(opts.get_name(), name);
+    /// ```
+    pub fn set_name(&mut self, name: &str) {
+        set_name(self.opts.pin_mut(), name);
+    }
+
+    /// Sets the `Options`' description.
+    ///
+    /// # Examples
+    ///
+    ///```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let mut opts = Options::new();
+    /// let description = "I love Seastar";
+    /// opts.set_description(description);
+    ///
+    /// assert_eq!(opts.get_description(), description);
+    /// ```
+    pub fn set_description(&mut self, description: &str) {
+        set_description(self.opts.pin_mut(), description);
+    }
+
+    /// Sets the `Options`' number of threads.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let mut opts = Options::new();
+    /// let smp = 42;
+    /// opts.set_smp(smp);
+    ///
+    /// assert_eq!(opts.get_smp(), smp);
+    /// ```
+    pub fn set_smp(&mut self, smp: u32) {
+        set_smp(self.opts.pin_mut(), smp);
+    }
+}
+
+impl Default for Options {
+    fn default() -> Self {
+        Options::new()
+    }
+}
+
+/// The object through which the contents of a `main` function would be ran in a Seastar app.
+/// Configurable through [`Options`].
+pub struct AppTemplate {
+    app: UniquePtr<app_template>,
+}
+
+impl AppTemplate {
+    /// Creates a Seastar app based on its configuration (`[seastar-rs::AppTemplate`]).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::{AppTemplate, Options};
+    ///
+    /// let app = AppTemplate::new_from_options(Options::default());
+    /// ```
+    pub fn new_from_options(mut opts: Options) -> Self {
+        AppTemplate {
+            app: new_app_template_from_options(opts.opts.pin_mut()),
+        }
+    }
+
+    /// Runs an app with a void callback (the output of which is always 0) and program arguments (argv).
+    /// The app is run on a separate thread.
```

Comment:
When I read this my understanding was that `run_void` would start a new thread for me and run the app there. 
Let's change it to something like:
`Each call to run_void must be done on a new thread`, or something like that (?)